lib/permit/types.ex

```diff
@@ -18,6 +18,9 @@ defmodule Permit.Types do
           | {atom(), any()}
           | (struct() -> boolean())
           | (Types.subject(), struct() -> boolean())
+          | {(struct() -> boolean()), (struct() -> Ecto.Query.t())}
+          | {(Types.subject(), struct() -> boolean()),
+             (Types.subject(), struct() -> Ecto.Query.t())}
```

Comment:
In `condition.ex` we have this as the type of the `:condition` key:
```elixir
          condition:
            boolean()
            | {atom(), (struct(), struct() -> any())}
            | (struct(), struct() -> boolean())
            | (struct() -> boolean()),
```
Some parts of this are matching, some are not. I'd like us to be able to have it delegated to `condition: Types.condition()`  in `condition.ex`, and also for this to be a bit more descriptive in some cases.
For example, `subject()` is a `struct()`, but `subject` describes it better in our domain. I'd like to see a similar clarification on what an `atom` is in this context (I think it's an action name but I'm not sure - generally atom could be anything, even a module), as well as what a `struct` is (is it the "object"?)